src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/AesCipherTests.cs

```diff
@@ -608,16 +609,42 @@ private static void RandomKeyRoundtrip(Aes aes)
 
         private static byte[] AesEncryptDirectKey(Aes aes, byte[] key, byte[] iv, byte[] plainBytes)
         {
-            using (MemoryStream output = new MemoryStream())
-            using (CryptoStream cryptoStream = new CryptoStream(output, aes.CreateEncryptor(key, iv), CryptoStreamMode.Write))
+            try
             {
-                cryptoStream.Write(plainBytes, 0, plainBytes.Length);
-                cryptoStream.FlushFinalBlock();
+                using (MemoryStream output = new MemoryStream())
+                using (CryptoStream cryptoStream = new CryptoStream(output, aes.CreateEncryptor(key, iv), CryptoStreamMode.Write))
+                {
+                    cryptoStream.Write(plainBytes, 0, plainBytes.Length);
+                    cryptoStream.FlushFinalBlock();
 
-                return output.ToArray();
+                    return output.ToArray();
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception(
+                    $"key.Length = {key.Length};\n"
+                    + $"iv.Length = {iv.Length};\n"
+                    + $"aes.KeySize = {aes.KeySize};\n"
+                    + $"aes.GetType().FullName = {aes.GetType().FullName};\n"
+                    + $"aes.ValidKeySize(192)? = {aes.ValidKeySize(192)};\n"
```

Comment:
`ValidKeySize(192)` is not useful. @bartonjs - do we want any additional diagnostics in here?